mono/metadata/class.c

```diff
@@ -48,6 +48,8 @@
 #include <mono/utils/bsearch.h>
 #include <mono/utils/checked-build.h>
 
+#define ENABLE_LAZY_ARRAY_IFACES TRUE
```

Comment:
Do you intend to leave this in as a feature flag? Most of the code doesn’t check it; also test with `FALSE`/undef.